docs/architecture/adr-001-autopilot-oci-basic-auth-support.md

```diff
@@ -110,6 +110,39 @@ secretRef:
   name: artifacts-token-based-auth
 ```
 
+#### Expected workflow
+
+This feature supports pulling both the k0s binary and the k0s air-gap bundle.
+Each repository tag should host only one artifact. To remotely host an artifact
+and later retrieve it using this feature, the following command is sufficient:
+
+```sh
+oras push <registry-address>/<namespace>/<repository>:<tag> <artifact path>
+```
+
+For example, to push the k0s binary and its corresponding airgap bundle, use
+the following commands:
+
+```sh
+# pushing the k0s binary.
+oras push registry.company.com/k0s/binary:v1.31.1 k0s
+
+# pushing the k0s airgap bundle.
+oras push registry.company.com/k0s/bundle:v1.31.1 k0s-airgap-bundle
+```
+
+Once the artifacts are hosted in the OCI registry, the following configuration
+is sufficient to enable the pull through OCI:
+
+```yaml
+url: oci://registry.company.com/k0s/binary:v1.31.1
+sha256: <sha-for-the-binary>
```

Comment:
Big question: Is this the digest of the image or the digest of the embedded binary?

When dealing with OCI artifacts, it would seem more natural to be the digest of the image (or image index) itself.

If it is the latter, this means that as long as the published artifact also uses SHA-256 as its digest algorithm (I have yet to see an image that doesn't), we would have an accurate way of figuring out which artifact to pick out of the image index. The downside would be that this would make multi-platform downloads with the same name impossible.